rs/artifact_pool/src/ingress_pool/peer_counter.rs

```diff
@@ -0,0 +1,219 @@
+use std::collections::{btree_map::Entry, BTreeMap};
+
+use ic_interfaces::ingress_pool::IngressPoolObject;
+use ic_logger::{warn, ReplicaLogger};
+use ic_types::{CountBytes, NodeId};
+
+#[derive(Clone)]
+/// Keeps track of the number of messages and total size of the messages per peer.
+pub(super) struct PeerCounters {
+    bytes_counters: PeerCounter,
+    count_counters: PeerCounter,
+}
+
+impl PeerCounters {
+    /// Create a new [`PeerCounters`] without any limits on the number/size of messages.
```

Comment:
when do we need this peer counter without limits?

Reply:
For the unvalidated section of the ingress pool